File: Core/Entities/EntityImage.cs
using System;

namespace Core.Entities
{
    public class EntityImage : BaseEntity
    {
        public string RelativePath { get; set; } = string.Empty;
        public string OriginalFileName { get; set; } = string.Empty;
        public string MimeType { get; set; } = string.Empty;
        public long FileSize { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: client/src/app/shared/models/entity-image.ts
export type EntityType =
  | 'Company'
  | 'Brand'
  | 'ElectricBikeProduct'
  | 'AgriculturalMachineProduct'
  | 'Product'
  | 'ProductBrand'
  | 'ProductType'
  | 'Order'
  | 'DeliveryMethod'
  | 'User';

export interface EntityImage {
  id: string;
  url: string;
  originalFileName: string;
  mimeType: string;
  fileSize: number;
  isUsed: boolean;
  createdAt: string;
}
